src/app/my-messages/todo-notification/todo-notification.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import { CommunityRegistrationService } from '../../shared/community-registration.service';
+import { HttpErrorResponse } from '@angular/common/http';
+import { DeleteConfirmationDialogComponent } from '../../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { CommunityUserRegistration } from '../../shared/community-user-registration';
+import { NotificationType } from '../notification-type.enum';
+import { TodoNotification } from './todo-notification';
+
+@Component({
```

Comment:
As far as I see the `TodoNotificationComponent` is used primarily to provide the implementation of some common methods. It does not have any template and cannot be instantiated independently from other components. So I suggest to remove `@Component` directive from the class, remove `OnInit` interface and make the class abstract. Perhaps it would be good to rename the class into something like `TodoNotificationComponentTrait`. Here the word `Trait` is used in the same meaning as in Scala language (see https://docs.scala-lang.org/tour/traits.html). It is an interface which can be partially implemented (these are the interfaces with default implementation in Java). What do you think about it?